### PR TITLE
update to Swift 3, Xcode 8, Cocoapods 1.1.1

### LINKED ISSUE #31: Add support for newer libraries in Xcode 8
```
import XCPlaygroundPage
XCPlaygroundPage.currentPage.needsIndefiniteExecution = true
```

is replaced by

```
import PlaygroundSupport
PlaygroundPage.current.needsIndefiniteExecution = true
```

> This change was mentioned in [Session 213 at WWDC 2016](https://developer.apple.com/videos/play/wwdc2016/213/).

[Reference](http://stackoverflow.com/a/37864618/159602)

### PR DESCRIPTION
**Pull request by askielboe**
This fixes #31, #38 & #43 for me using Xcode 8, Swift 3 and Cocoapods 1.1.1.

**Note that this will most likely break support for Swift 2 and Xcode 7!**

So a better solution is probably needed. I also have not tested this using Carthage.

Just putting it here in case anyone needs it.

### CODE REVIEW

**Comment by edasque** [comment]
That worked for me with `pod playgrounds Alamofire`. Can't attest to what happens in Xcode 7.

File: lib/cocoapods-playgrounds/workspace.rb [hunk 1]
```diff
@@ -168,8 +169,8 @@ def generate_project
     def generate_swift_code(path)
       File.open(path + 'Contents.swift', 'w') do |f|
         f.write("//: Please build the scheme '#{target_name}' first\n")
-        f.write("import XCPlayground\n")
```
Comment (segiddins) [inline lib/cocoapods-playgrounds/workspace.rb:171]: Could this use `if (swift>=...)`?

Comment (askielboe) [reply lib/cocoapods-playgrounds/workspace.rb:171]: Yes definitely. I only made this as a quick fix. I will update to include some swift version #ifdefs.

**Comment by amsmonteiro** [comment]
@askielboe everything seems to work ok! we just need swift versions conditionals :)

**Comment by askielboe** [comment]
yea, sorry I haven't taken time to look at this. I'm not sure what the best way to implement an optional swift version is.

**Comment by amsmonteiro** [comment]
@askielboe Perhaps there's a clever way to do this but a simple if would suffice:
`#if swift(>=3.0)
        f.write("import PlaygroundSupport\n")
  #else
        f.write("import XCPlayground\n")
  #endif`

**Comment by edasque** [comment]
Can we merge this, at least into the branch? I can't figure out how I was able to get this working initially and am not a Ruby expert.

**Comment by edasque** [comment]
@askielboe can you add the swift conditional, I think that's the only thing preventing this from being merged. Let us all enjoy your fix!

**Comment by jawwad** [comment]
@askielboe - One idea could be to add a command line option such as "--use-legacy-swift" or "--use-legacy-xcode", that if passed would build the workspace the old way.

@segiddins - Do you feel that this gem still needs to support Swift 2 and Xcode 7? Xcode 8 has been out of beta for a while now, and it's also [not possible](http://stackoverflow.com/a/38326852/340508) to create a playground for Swift 2 with Xcode 8.

**Comment by segiddins** [comment]
Yes, I feel like we should support Xcode 7 as well, no sense breaking backwards compatibility when we don't have to

**Comment by jawwad** [comment]
@askielboe - I know you are probably busy and haven't gotten a chance to look at this again. Do you think you'll have the time to modify your PR to include a legacy flag to support Xcode 7?

**Comment by jawwad** [comment]
@segiddins - It's been about two months since we've last heard from @askielboe regarding this PR. What can we do here? Would really like to get this working with the current version of Xcode. Suggestions?

**Comment by segiddins** [comment]
@jawwad feel free to open a new PR based on this one

**Comment by jawwad** [comment]
@segiddins - Yeah, I'd love to take a look, but I'm a bit pressed for time until around March/April. If this is still open I'll try taking a look at it then.

**Comment by edasque** [comment]
Well, this repo is now getting a lot of traffic but there is still no update to be able to use it with the most recent Swift & Xcode.

**Comment by bleeckerj** [comment]
@edasque I've applied this patch (https://github.com/segiddins/ThisCouldBeUsButYouPlaying/pull/44/files/60008f138de69c00540953c193493062a3742a48) and it works fine. It's a pull that just hasn't been brought into the main branch, but you should be able to apply it yourself.

**Comment by edasque** [comment]
@bleeckerj do you mind detailing how to apply it? I am not sure how to do that with a globally installed gem with a binary.

**Comment by bleeckerj** [comment]
Sure. Your mileage may vary, but I simply found the workspace.rb file and made the changes by hand, I believe there are maybe four diffs.

To find where your gems are, run 

`gem env`

and look for INSTALLATION_DIRECTORY. Navigate there and look for `cocoapods-playgrounds-1.2.2` (in my case on macOS it was in `/Library/Ruby/Gems/2.0.0/gems/cocoapods-playgrounds-1.2.2/lib/cocoapods-playgrounds`)

In there should be `workspace.rb` which is the file you need to patch. Like I said, Id did it by hand.

In any case, what you need to do is change workspace.rb in some fashion (you could clone the repo and apply the patch there and then install it) and make sure it lives where ruby expects to pick up your gems.

**Comment by edasque** [comment]
Didn't seem to work for me. ```The file /private/tmp/SwifterSwift.xcworkspace does not exist.```

**Comment by xcadaverx** [comment]
@edasque you can fix that error by ignoring line 80 on the patch he linked to. Just leave the `target_dir +` as it was. You really only need to apply line 139.

**Comment by neonichu** [comment]
Sorry, the project isn't being maintained anymore at this point.

**Pull request closed without merging**